src/Tests/With_UTF8_BOM.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System.IO;
+using System.Text;
+using NServiceBus.MessageInterfaces.MessageMapper.Reflection;
+using NServiceBus.Newtonsoft.Json;
+using NUnit.Framework;
+
+[TestFixture]
+public class With_UTF8_BOM
```

Comment:
Let's rename this to `When_deserializing_a_message`

Reply:
Done.